OptionSelector/AnimalTypesTableViewController.m

```diff
@@ -0,0 +1,178 @@
+//
+//  AnimalTypesTableViewController.m
+//  OptionSelector
+//
+//  Created by Lauren Caponong on 8/11/15.
+//  Copyright (c) 2015 Mike Kavouras. All rights reserved.
+//
+
+#import "AnimalTypesTableViewController.h"
+#import "AnimalDetailTableViewController.h"
+#import "CQCategory.h"
+
+@interface AnimalTypesTableViewController ()  <ViewControllerAnimalDetailTableViewDelegate>
+
+@property (nonatomic) NSMutableArray *array;
+@property (nonatomic) NSInteger index;
+
+@end
+
+
+
+
+@implementation AnimalTypesTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    
+    
+    // initialize the array to hold our types - dog, cat, bug
+    
+    self.array = [[NSMutableArray alloc]init];
+
+    
+    
+    // creating instances of CQCategory and initializing them
+    
+    CQCategory *dog = [[CQCategory alloc] init];
+    CQCategory *cat = [[CQCategory alloc] init];
+    CQCategory *bug = [[CQCategory alloc] init];
+    
+    
+    
+    // the 'name' property of the dog, cat, bug instances of CQCategory will be those strings
+    
+    dog.name = @"Dog";
+    cat.name = @"Cat";
+    bug.name = @"Bug";
+    
+    
+    
+    // breeds of dog - these are our options to choose from
+    
+    dog.options = @[
+                           @"Golden Retriever",
+                           @"Yorkie",
+                           @"Alaskan Malamute",
+                           @"Pug",
+                           @"Bulldog",
+                           @"Bernese Mountain Dog",
+                           @"Labrador Retriver",
+                           @"Mastiff"];
+    dog.selection = @" ";
+    
+    
+    
+    
+    
+    // breeds of cat - these are our options to choose from
+    
+    cat.options = @[
+                           @"Russian Blue",
+                           @"Calico",
+                           @"Siamese",
+                           @"British Shorthair",
+                           @"Persian",
+                           @"Cornish Rex",
+                           @"Scottish Fold"];
+    
+    cat.selection = @" ";
+    
+    
+    
+    
+    
+    // breeds of bug - these are our options to choose from
+    
+    bug.options = @[
+                           @"Butterfly",
+                           @"Ladybug",
+                           @"Caterpillar",
+                           @"Spider",
+                           @"Dragonfly",
+                           @"Cricket"];
+    
+    bug.selection = @" ";
+    
+    
+    
+    // adding these CQCategory instancse - dog, cat, bug - to our array to hold the 'types' of animal
+    
+    [self.array addObject:dog];
+    [self.array addObject:cat];
+    [self.array addObject:bug];
```

Comment:
Could you move your data-building code to its own method so that it's better encapsulated? Something like `loadAnimalCategories`?